test-control/src/main/java/com/pkb/common/testcontrol/services/DefaultProcessingControllerService.java

```diff
@@ -46,6 +47,9 @@ private void applyToRoutesExceptTestControl(ThrowingConsumer<Route, Exception> r
         if (reverseOrder) {
             comp = comp.reversed();
         }
+        // don't let camel wait gracefully to shut down processing as this can time out tests (and there's no point in
+        // waiting for anything at the point of calling suspend in test-control)
+        context.getShutdownStrategy().setTimeout(1);
```

Comment:
it should wait for pending operations to finish, shouldn't it? otherwise we risk leaking state again.

Reply:
This configures the DefaultShutdownStrategy's timeout, which is the timeout it waits until it tries to shutdown _hard_. (See https://camel.apache.org/manual/graceful-shutdown.html)

KMS is having lots of timeouts in e2e tests because of the default 45 sec graceful waits. At the point of test-control calling suspending I think we shouldn't be waiting for anything and it's okay to kill things because everything should be asserted by now.